### PR TITLE
Fix db lock

### PR DESCRIPTION
**Pull request by tharvik**
PR to find out more about this "db lock" issue.

It seems that when loading the ArtiConfig, the same cache path was used, so getting rid of it and using the parametrized cache path fixes it.

I also added some improvements on the github hooks, hopefully accelerating and simplifying it

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -11,16 +11,9 @@ jobs:
...
-      - uses: actions/cache@v2
-        with:
-          path: |
-            ~/.cargo/registry
-            ~/.cargo/git
-            target
-          key: ${{ runner.os }}-build-android-${{ hashFiles('**/Cargo.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-build-android
+      - uses: Swatinem/rust-cache@v1
```
Comment (ineiti) [inline .github/workflows/test.yaml:14]: What is the advantage of rust-cache here? Reading its github-page I'm not sure if it also caches the compiled files or just the downloaded ones.

Comment (tharvik) [reply .github/workflows/test.yaml:19]: It is more readable and better caches than `actions/cache`. It reduce [greatly the test time](https://github.com/c4dt/arti-rest/runs/2885756369).

Comment (ineiti) [reply .github/workflows/test.yaml:19]: I think the other cache only saved when the workflow passed or something like that. It already reduced the test time a lot, but only after a successful test.

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -11,39 +11,23 @@ jobs:
...
-          ln -s $ANDROID_NDK_HOME/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android30-clang $HOME/bin/aarch64-linux-android-clang
-          export PATH=$PATH:$HOME/bin:$ANDROID_NDK_HOME/toolchains/llvm/prebuilt/linux-x86_64/bin
-          export CARGO_TARGET_AARCH64_LINUX_ANDROID_LINKER=${ANDROID_NDK_HOME}/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android30-clang
-          cargo build --target aarch64-linux-android
-        env:
-          ANDROID_NDK_HOME: ${{ steps.setup-ndk.outputs.ndk-path }}
+      - uses: actions-rs/cargo@v1
+        with:
+          command: ndk
+          args: --target aarch64-linux-android build
```
Comment (cgrigis) [inline .github/workflows/build.yaml:30]: :heart:

**Review by cgrigis — APPROVED** [review]
LGTM

**Pull request merged**